File: 2050-count-good-numbers/2050-count-good-numbers.go

const MOD int64 = int64(1e9) + 7

func bpow(base, exp int64) int64 {
	res := int64(1)
	for exp > 0 {
		if exp%2 == 1 {
			res = (res * base) % MOD
			exp--
		}
		base = (base * base) % MOD
		exp >>= 1
	}
	return res
}
func countGoodNumbers(n int64) int {
	return int((bpow(4, n/2) * bpow(5, n/2+n%2)) % MOD)
}

File: 2050-count-good-numbers/2050-count-good-numbers.cpp
using ll = long long;
class Solution {
   public:
    const int MOD = 1e9 + 7;
    ll bpow(ll base, ll exp) {
        ll res = 1;
        while (exp > 0) {
            if (exp % 2) {
                res = (res * base) % MOD;
                exp--;
            }
            exp >>= 1;
            base = (base * base) % MOD;
        }
        return res;
    }
    int countGoodNumbers(long long n) {
        ll numOdd = n / 2;
        ll numEven = n / 2 + (n % 2);

        return (bpow(5, numEven) * bpow(4, numOdd)) % MOD;
    }
};